### DISCUSSION TITLE
Encoded SVG isn't working with arbitrary values

### DISCUSSION THREAD

**Discussion by Mr-Zafar** [tailwindlabs/tailwindcss#7079] [Help] [answered]
Encoded SVG isn't working with .bg-[url()] arbitrary values

&lt;ul role="list" class="space-y-2 text-sm"&gt;
  &lt;li class="before:inline-block before:w-[.8125rem] before:h-[.8125rem] before:bg-[length:.8125rem_.8125rem] before:bg-no-repeat before:bg-[right_center] before:bg-[url('data:image/svg+xml,%3Csvg width='10' height='8' viewBox='0 0 10 8' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M9.207 0.793C9.39447 0.980528 9.49978 1.23484 9.49978 1.5C9.49978 1.76516 9.39447 2.01947 9.207 2.207L4.207 7.207C4.01947 7.39447 3.76516 7.49979 3.5 7.49979C3.23483 7.49979 2.98052 7.39447 2.793 7.207L0.792995 5.207C0.610837 5.0184 0.510043 4.7658 0.512321 4.5036C0.5146 4.2414 0.619769 3.99059 0.805177 3.80518C0.990585 3.61977 1.2414 3.5146 1.50359 3.51233C1.76579 3.51005 2.01839 3.61084 2.20699 3.793L3.5 5.086L7.793 0.793C7.98052 0.605529 8.23483 0.500214 8.5 0.500214C8.76516 0.500214 9.01947 0.605529 9.207 0.793Z' fill='currentColor'/%3E%3C/svg%3E%0A')] before:fill-gray-500"&gt;Text goes here&lt;/li&gt;
  &lt;li class="before:inline-block before:w-[.8125rem] before:h-[.8125rem] before:bg-[length:.8125rem_.8125rem] before:bg-no-repeat before:bg-[right_center] before:bg-[url('data:image/svg+xml,%3Csvg width='10' height='8' viewBox='0 0 10 8' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M9.207 0.793C9.39447 0.980528 9.49978 1.23484 9.49978 1.5C9.49978 1.76516 9.39447 2.01947 9.207 2.207L4.207 7.207C4.01947 7.39447 3.76516 7.49979 3.5 7.49979C3.23483 7.49979 2.98052 7.39447 2.793 7.207L0.792995 5.207C0.610837 5.0184 0.510043 4.7658 0.512321 4.5036C0.5146 4.2414 0.619769 3.99059 0.805177 3.80518C0.990585 3.61977 1.2414 3.5146 1.50359 3.51233C1.76579 3.51005 2.01839 3.61084 2.20699 3.793L3.5 5.086L7.793 0.793C7.98052 0.605529 8.23483 0.500214 8.5 0.500214C8.76516 0.500214 9.01947 0.605529 9.207 0.793Z' fill='currentColor'/%3E%3C/svg%3E%0A')] before:fill-gray-500"&gt;Text goes here&lt;/li&gt;
&lt;/ul&gt;

It only works if you call a file

![image](https://user-images.githubusercontent.com/5662351/149616466-ce04ce48-2d62-4d8c-a7a5-28c552d8ba2b.png)

like that:

&lt;ul role="list" class="space-y-2 text-sm"7gt;
  &lt;li class="before:inline-block before:w-[.8125rem] before:h-[.8125rem] before:bg-[length:.8125rem_.8125rem] before:bg-no-repeat before:bg-[right_center] before:bg-[url('../svg/component/check.svg')] before:mr-2"7gt;Text goes here&lt;/li7gt;
  &lt;li class="before:inline-block before:w-[.8125rem] before:h-[.8125rem] before:bg-[length:.8125rem_.8125rem] before:bg-no-repeat before:bg-[right_center] before:bg-[url('../svg/component/check.svg')] before:mr-2"7gt;Text goes here&lt;/li7gt;
&lt;/ul7gt;

**Comment by adamwathan** [comment, marked as answer]
Problem is all of the spaces, you can't have spaces in class names. You're probably better off base 64 encoding it, or alternatively going through and adding an underscore for every space.
